nhais-adaptor/edifact/models/message.py

```diff
@@ -0,0 +1,155 @@
+            Segment(key="NAD", value=f"FHS+{party_id}:954"),
+            Segment(key="DTM", value=f"137:{formatted_date_time}:203"),
+            Segment(key="RFF", value=f"950:{ref_number}"),
+        ]
+        super().__init__(segments=segments)
+
+
+class MessageSegmentTrigger1(SegmentCollection):
+    """
+    A collection of segments to provide registration information for GP patients.
+    """
+
+    def __init__(self, transaction_number, party_id, acceptance_code, acceptance_type, date_time, location):
+        """
+        :param transaction_number: a unique transaction number. NHAIS will reference this in its response
+        :param party_id: GMC National code and the Local GP Code of the patient's GP (separated by “,”).
+        :param acceptance_code: The acceptance code "A" for Acceptance
+        :param acceptance_type: The acceptance type "1" for a Birth
+        :param date_time: date of the registration
+        :param location: the patients place of birth
+        """
+        formatted_date_time = datetime.strptime(date_time, '%Y-%m-%d %H:%M:%S.%f').strftime('%Y%m%d')
+        segments = [
+            Segment(key="S01", value="1"),
+            Segment(key="RFF", value=f"TN:{transaction_number}"),
+            Segment(key="NAD", value=f"GP+{party_id}:900"),
+            Segment(key="HEA", value=f"ACD+{acceptance_code}:ZZZ"),
+            Segment(key="HEA", value=f"ATP+{acceptance_type}:ZZZ"),
+            Segment(key="DTM", value=f"956:{formatted_date_time}:102"),
+            Segment(key="LOC", value=f"950+{location}"),
+        ]
+        super().__init__(segments=segments)
+
+
+class MessageSegmentTrigger2(SegmentCollection):
+    """
+    A collection of segments that represent personal information about a patient.
+    """
+
+    def __init__(self, id_number, name, date_of_birth, gender, address):
+        """
+        :param id_number: OPI official Payment Id (existing NHS Number)
+        :param name: the name of the patient
+        :param date_of_birth: Patients date of birth
+        :param gender: sex of the patient. For an acceptance transaction, reference "G1", this segment is required
+        :param address: the patients address
+        """
+        formatted_date = datetime.strptime(date_of_birth, '%Y-%m-%d').strftime('%Y%m%d')
+        segments = [
+            Segment(key="S02", value="2"),
+            Segment(key="PNA",
+                    value=f"PAT+{id_number}:OPI+++SU:{name.family_name}+FO:{name.first_given_forename}+TI:{name.title}+MI:{name.middle_name}+FS:{name.third_given_forename}"),
+            Segment(key="DTM", value=f"329:{formatted_date}:102"),
+            Segment(key="PDI", value=f"{gender}"),
+            Segment(key="NAD",
+                    value=f"PAT++{address.house_name}:{address.address_line_1}:{address.address_line_2}:{address.town}:{address.county}+++++{address.post_code}")
+        ]
+        super().__init__(segments=segments)
+

```

Comment:
Seems like some of these keys are being used multiple times ("NAD"?), could they be extracted into human readable variable names? Just to avoid having a lot of repeated magic strings

Reply:
I think so but I'm little inclined to doing this at the moment. When I come to build the parser of the edifact message this will probably change so the parser will need to know what it is working with. So constants and keys will be generated then

Reply:
That's fine, as long as it's something we're aware of.